cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -731,28 +730,70 @@ public TSStatus executeNonQueryPlan(ExecutNonQueryReq request)
   public void syncLeaderWithConsistencyCheck(boolean isWriteRequest)
       throws CheckConsistencyException {
     if (isWriteRequest) {
-      if (!syncLeader()) {
-        throw CheckConsistencyException.CHECK_STRONG_CONSISTENCY_EXCEPTION;
-      }
+      syncLeader(new StrongCheckConsistency());
     } else {
-      switch (ClusterDescriptor.getInstance().getConfig().getConsistencyLevel()) {
+      switch (config.getConsistencyLevel()) {
         case STRONG_CONSISTENCY:
-          if (!syncLeader()) {
-            throw CheckConsistencyException.CHECK_STRONG_CONSISTENCY_EXCEPTION;
-          }
+          syncLeader(new StrongCheckConsistency());
           return;
         case MID_CONSISTENCY:
-          // do not care success or not
-          syncLeader();
+          // if leaderCommitId bigger than localAppliedId a value,
+          // will throw CHECK_MID_CONSISTENCY_EXCEPTION
+          syncLeader(
+              new CheckConsistency() {
```

Comment:
why not provide a midCheckConsistency like strongCheckConsistency?

Reply:
strongcheckConsistency have two place using, due to Code Reusability..
midcheckconsistency only use in a place,  do not extracts a class, for Code readability..

Reply:
But I do not think it is necessary to `new` each time the method is called, as they are not changed throughout the process. Maybe you can make them static members of a class.

Reply:
+1 for jt2594838